src/button.js

```diff
@@ -0,0 +1,92 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+
+import {AUTOCOMPLETE_CONTEXT} from './constants'
+import {compose} from './utils'
+
+class Button extends Component {
+  static contextTypes = {
+    [AUTOCOMPLETE_CONTEXT]: PropTypes.object.isRequired,
+  }
+
+  static propTypes = {
+    disabled: PropTypes.bool,
+    onKeyDown: PropTypes.func,
+    onClick: PropTypes.func,
+  }
+
+  constructor(props, context) {
+    super(props, context)
+    this.autocomplete = this.context[AUTOCOMPLETE_CONTEXT]
+    this.handleKeyDown = compose(this.handleKeyDown, props.onKeyDown)
+    this.handleClick = compose(this.handleClick, props.onClick)
+  }
+
+  keyDownHandlers = {
+    ArrowDown(event) {
```

Comment:
Yes :) I had that working for the Label component and then I started making this one 😆 I'll add it here.